Cython/Compiler/Code.py

```diff
@@ -1215,6 +1215,13 @@ def initialize_main_c_code(self):
         code.putln("")
         code.putln("/* --- Runtime support code --- */")
 
+    def initialize_main_h_code(self):
+        # h code can include utility code in a few cases so it needs to be insertable
+        rootwriter = self.rootwriter
+        for part in self.code_layout:
+            if part.find("utility_code") != -1:
```

Comment:
It took me a while to recognise this as a verbose spelling of `"utility_code" in part`. I'd rather use
```suggestion
            if part.startswith("utility_code"):
```
However, DRY aside, I wonder if we should move the implementation parts further down in the header file, as we do in the C file. In fact, I wonder if we should allow an implementation in a header file at all… There is a `proto_before_types` for a reason, and the same applies to the non-`before_types` section. Could we only allow the two `utility_code_proto` sections for now, and move them before and after the type declarations?